### ISSUE TITLE
about "the forward pass is faster on the GPU whereas the loss is often (depending on your hyperparameters) faster on the CPU"

### ISSUE THREAD

**Issue by sunnnnnnnny** [CorentinJ/Real-Time-Voice-Cloning#44]
Why is loss calculated faster on the cpu, and what parameters are you depending on?thanks

**Comment by CorentinJ** [comment]
The loss was faster on CPU before it was properly vectorized, because it was a lot of back-and-forth. The parameters that affect this are the number of speakers per batch and the number of utterances per speaker. Now the loss should be just as fast on GPU, and for the sake of simplicity it would be much better to only have one device, but for reasons unknown the loss is incorrect (it is `None` IIRC) when computed on GPU, so I've left it to be computed on the CPU while the forward pass (a recurrent neural network) is much faster on GPU. I still have to figure out what that bug is.
